spark/src/main/scala/com/cloudera/livy/spark/SparkManager.scala

```diff
@@ -91,9 +101,25 @@ private class SparkYarnManager(processFactory: SparkProcessBuilderFactory) exten
 
   val interactiveManager = new SessionManager(processFactory.livyConf, interactiveFactory)
 
+  def clientManager: SessionManager[ClientSession, CreateClientRequest] = {
+    throw new UnsupportedOperationException
+  }
+
   override def shutdown(): Unit = {
     batchManager.shutdown()
     interactiveManager.shutdown()
     client.close()
   }
-}
\ No newline at end of file
+}
+
+private class SparkClientManager(processFactory: SparkProcessBuilderFactory)
```

Comment:
I'm a little confused about these changes; what do they have to do with file uploads? Why are they needed? Why is `SparkClientManager` inheriting from `SparkYarnManager` since it's not necessarily YARN-related?